url7f729337e3_timmolter_Sundial_tgz-pJ8-org_knowm_sundial_SampleRunJ8/src/org/quartz/plugins/xml/XMLSchedulingDataProcessor.java

```diff
@@ -0,0 +1,592 @@
+   */
+  private static final String XSD_DATE_FORMAT = "yyyy-MM-dd'T'hh:mm:ss";
+
+  private static final SimpleDateFormat dateFormat = new SimpleDateFormat(XSD_DATE_FORMAT);
+
+  /*
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Data members.
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   */
+
+  // scheduling commands
+  private List<JobDetail> loadedJobs = new LinkedList<JobDetail>();
+  private List<OperableTrigger> loadedTriggers = new LinkedList<OperableTrigger>();
+
+  private Collection<Exception> validationExceptions = new ArrayList<Exception>();
+
+  private ClassLoadHelper classLoadHelper = null;
+
+  private DocumentBuilder docBuilder = null;
+  private XPath xpath = null;
+
+  private final Logger logger = LoggerFactory.getLogger(XMLSchedulingDataProcessor.class);
+
+  /*
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Constructors.
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   */
+
+  /**
+   * Constructor for JobSchedulingDataLoader.
+   *
+   * @param classLoadHelper
+   * @param clh class-loader helper to share with digester.
+   * @throws ParserConfigurationException if the XML parser cannot be configured as needed.
+   */
+  public XMLSchedulingDataProcessor(ClassLoadHelper classLoadHelper) throws ParserConfigurationException {
+
+    this.classLoadHelper = classLoadHelper;
+
+    initDocumentParser();
+  }
+
+  /**
+   * Initializes the XML parser.
+   *
+   * @throws ParserConfigurationException
+   */
+  private void initDocumentParser() throws ParserConfigurationException {
+
+    DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+
+    docBuilderFactory.setNamespaceAware(false);
+    docBuilderFactory.setValidating(true);
+
+    docBuilderFactory.setAttribute("http://java.sun.com/xml/jaxp/properties/schemaLanguage", "http://www.w3.org/2001/XMLSchema");
+
+    docBuilderFactory.setAttribute("http://java.sun.com/xml/jaxp/properties/schemaSource", resolveSchemaSource());
+
+    docBuilder = docBuilderFactory.newDocumentBuilder();
+
+    docBuilder.setErrorHandler(this);
+
+    xpath = XPathFactory.newInstance().newXPath();
+  }
+
+  private Object resolveSchemaSource() {
+
+    InputSource inputSource = null;
+
+    InputStream is = null;
+
+    // try {
+    is = classLoadHelper.getResourceAsStream(QUARTZ_XSD_PATH_IN_JAR);
+
+    if (is == null) {
+      logger.warn("Could not load jobs schema from classpath!");
+    } else {
+      inputSource = new InputSource(is);
+    }
+
+    return inputSource;
+  }
+
+  /**
+   * Process the xml file in the given location, and schedule all of the jobs defined within it.
+   *
+   * @param fileName meta data file name.
+   */
+  public void processFile(String fileName, boolean failOnFileNotFound) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.